keps/sig-storage/1472/README.md

```diff
@@ -0,0 +1,1416 @@
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+There are two types of volumes that are getting created after making a scheduling
+decision for a pod:
+- [ephemeral inline
+  volumes](https://kubernetes-csi.github.io/docs/ephemeral-local-volumes.html) -
+  a pod has been permanently scheduled onto a node
+- persistent volumes with [delayed
+  binding](https://kubernetes.io/docs/concepts/storage/storage-classes/#volume-binding-mode)
+  (`WaitForFirstConsumer`) - a node has been selected tentatively
+
+In both cases the Kubernetes scheduler currently picks a node without
+knowing whether the storage system has enough capacity left for
+creating a volume of the requested size. In the first case, `kubelet`
+will ask the CSI node service to stage the volume. Depending on the
+CSI driver, this involves creating the volume. In the second case, the
+`external-provisioner` will note that a `PVC` is now ready to be
+provisioned and ask the CSI controller service to create the volume
+such that is usable by the node (via
+[`CreateVolumeRequest.accessibility_requirements`](https://kubernetes-csi.github.io/docs/topology.html)).
+
+If these volume operations fail, pod creation may get stuck. The
+operations will get retried and might eventually succeed, for example
+because storage capacity gets freed up or extended. A pod with an
+ephemeral volume will not get rescheduled to another node. A pod with
+a volume that uses delayed binding should get scheduled multiple times,
+but then might always land on the same node unless there are multiple
+nodes with equal priority.
+
+A new API for exposing storage capacity currently available via CSI
+drivers and a scheduler enhancement that uses this information will
+reduce the risk of that happening.
+
+## Motivation
+
+### Goals
+
+* Define an API for exposing storage capacity information.
+
+* Expose capacity information at the semantic
+  level that Kubernetes currently understands, i.e. in a way that
+  Kubernetes can compare capacity against the requested size of
+  volumes. This has to work for local storage, network-attached
+  storage and for drivers where the capacity depends on parameters in
+  the storage class.
+
+* Increase the chance of choosing a node for which volume creation
+  will succeed by tracking the currently available capacity available
+  through a CSI driver and using that information during pod
+  scheduling.
+
+### Non-Goals
+
+* Only CSI drivers will be supported.
```

Comment:
The way this sentence is written, it is a "Goal", not "Non-Goal":).

Reply:
Updated. I always struggle with avoiding double-negation in this section.